### ISSUE TITLE
analysis_options exclude isn't working, behavior differs between flutter analyze and dartanalyzer

### ISSUE THREAD

**Issue by johnpryan** [dart-lang/sdk#38753]
Excluding files still displays analyzer warnings in IntelliJ and in dartanalyzer,  but not with `flutter analyze`

Here's my analysis_options.yaml file:

```yaml
analyzer:
  exclude:
    - lib/exclude_me/*.exclude.dart
  strong-mode:
    implicit-dynamic: false
```

Flutter 1.9.1+hotfix.4,
Dart SDK 2.5.0

## Steps to reproduce

1. clone https://github.com/johnpryan/analyzer_exclude_bug
2. run `flutter analyze` (no errors)
3. run `dartanalyzer .` 
4. observe dartanalyzer displays warnings, even  though ànalysis_options.yaml`  excludes the file.

**Comment by chalin** [comment]
Related: #25551. The now closed dup #34053, also listed these related items: #28463, #25551, #29430

**Comment by johnpryan** [comment]
Good find, thanks. I think this is a dupe of #25551
